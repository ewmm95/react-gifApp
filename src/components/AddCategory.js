import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = ({target: {value}}) => setInputValue(value)
    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <= 2) return;
        setCategories( x => [inputValue, ...x] );
        setInputValue('');
    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                className='input-search'
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}