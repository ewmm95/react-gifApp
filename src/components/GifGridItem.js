import React from 'react';
import 'animate.css';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
    return (
        <div className='card animate__animated animate__zoomIn' tabIndex="0">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}