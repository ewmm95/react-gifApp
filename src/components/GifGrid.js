import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import 'animate.css';

export const GifGrid  = ({category}) => {
    const { data: images, loading} = useFetchGifs(category);
    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <h3  className='animate__animated animate__flash'>Cargando...</h3>}
            <div className='cards'>
                { images.map( img => <GifGridItem key={img.id} {...img}/>) }
            </div>
        </>
    )
}
