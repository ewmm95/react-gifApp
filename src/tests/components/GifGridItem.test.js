import React from "react"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem />', () => {
    const props = {
        title: 'Titulo',
        url: 'https://localhost:180908/login',
    };
    const wrapper = shallow(<GifGridItem {...props}/>);
    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(props.title)
    })
    test('debe de tener una imagen con el title y la url', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(props.url);
        expect(img.prop('alt')).toBe(props.title);
    })
    test('debe de tener la clase animate__zoomIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__zoomIn')).toBe(true);
    })
    
})
