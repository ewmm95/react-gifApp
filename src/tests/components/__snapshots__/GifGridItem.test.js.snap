// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
  tabIndex="0"
>
  <img
    alt="Titulo"
    src="https://localhost:180908/login"
  />
  <p>
    Titulo
  </p>
</div>
`;
